Quote block: button for cite add/remove

ellatrix opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

All blocks with caption currently have a button to add/remove the caption. But not quote, which has caption-like logic. Let's do the same for caption and reuse the existing caption util.

Not sure if this helps with #15308.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

ellatrix committed:
Quote block: button for cite add/remove

ellatrix committed:
Same for pull quote

ellatrix committed:
Forgot default props

ellatrix commented:
> so this is a layout shift we should fix in this branch

Not sure why this would be an issue with this branch. Probably a styling issue with the inserter on quote block which seems out of scope for this PR. I don't see the current theme I have enabled.

Updating the icon now.

ellatrix committed:
Fix icon

ellatrix commented:
Icon changed.

mtias commented on packages/block-library/src/pullquote/edit.js:
```diff
@@ -66,32 +66,26 @@ function PullQuoteEdit( {
-							}
-						/>
-					) }
+					<Caption
```
Should we skip doing this for the pullquote block?

ellatrix committed:
Revert pullquote

ellatrix committed:
Fix e2e tests

mtias reviewed (approved):
Thanks for looking at this. This is feature that doesn't warrant being so visible, and it's nice to address that long standing problem of the hanging placeholder.

jasmussen commented:
> Not sure why this would be an issue with this branch. Probably a styling issue with the inserter on quote block which seems out of scope for this PR. I don't see the current theme I have enabled.

Seems fine to fix separately, but the the issue is invisible in trunk, but super visible here, so if we merge this one, we will need to fix the other issue shortly after.

ellatrix commented:
@jasmussen What if we just remove the appender? It doesn't seem so useful in the quote block. You can press enter and use the empty inserter.

ellatrix committed:
Fix tests

jasmussen commented:
> What if we just remove the appender? It doesn't seem so useful in the quote block. You can press enter and use the empty inserter.

IMO we need to track down why the quote block uses a different appender than the group block. It's outputting different CSS classes. As a container block of anything, it seems prudent to be consistent with other containers.

ellatrix committed:
Fix e2e

ellatrix committed:
Try to stabilise test

ellatrix committed:
Avoid native circular dependency

ellatrix committed:
Try again to stabilise test

dcalhoun commented:
👋🏻 @ellatrix. I noticed the mobile unit tests fail currently. It appears the https://github.com/WordPress/gutenberg/pull/59097 changes the were merged into this branch are no longer present. Was the removal an oversight during a force push?

ellatrix committed:
Re-merged change by @dcalhoun from #59097

ellatrix commented:
@dcalhoun Ah, thanks! I missed that. I re-applied your change but without the pull quote changes which were reverted for the web side as well.

dcalhoun committed:
test: Fix Rich Text failures (#60290)

* test: Define missing `logException` native module mock

Prevent cryptic test failure messages originating from invoking this
undefined method, which prevented the original error message from
surfacing in the test failure log.

* refactor: Export private API Rich Text module

This export was added for interoperability with the web version modified
in https://github.com/WordPress/gutenberg/pull/58916.

ellatrix committed:
Fix e2e test
